app/models/goal.py

```diff
@@ -1,5 +1,13 @@
 from app import db
 
-
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal")
+    
+    def goal_dict(self):
```

Comment:
While I think `goal_dict` is a fine name, it does have a bit of likely repetition since I imagine a lot of variables will also be named `goal` and thus you get `goal.goal_dict`. So I would prefer something like `to_dict`.

But this is a really minor nitpick.

Reply:
`to_dict` is especially a better name in this case also because it's what the similar method in the `Task` class below is called, so I would recommend `to_dict` here just to be consistent.